utils/misc.go

```diff
@@ -23,3 +26,57 @@ func GoWithRecovery(exec func(), recoverFn func(r interface{})) {
 	}()
 	exec()
 }
+
+// WithRetry provides a general retry logic.
+//
+// The given f will keep running, until:
+// - f returns true. It means work is done.
+// - retry times is greater than the given times
+// - ctx is done.
+//
+// Otherwise, this function will wait a time of the given duration
+// and continue to execute f.
+//
+// The argument provided for f is the retried times.
+func WithRetry(ctx context.Context, maxTimes uint, duration time.Duration, f func(uint) bool) {
```

Comment:
Can we add some unit tests for `WithRetry` and `WithRetryBackoff`? As util I think this is necessary~

Reply:
ok